File: frontend/lib/features/home/domain/entities/loan.dart
import 'package:equatable/equatable.dart';

class Loan extends Equatable {
  final String id;
  final String bookCode;
  final String studentId;
  final DateTime issueDate;
  final DateTime returnDate;
  final String status;

  const Loan({
    required this.id,
    required this.bookCode,
    required this.studentId,
    required this.issueDate,
    required this.returnDate,
    required this.status,
  });

  @override
  List<Object?> get props => [
        id,
        bookCode,
        studentId,
        issueDate,
        returnDate,
        status,
      ];
}

File: backend/library/models/loan.py
from pydantic import BaseModel
from typing import Optional
from bson import ObjectId
from fastapi.encoders import jsonable_encoder

from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class Loan(BaseModel):
    book_code: str
    student_id: str
    issue_date: datetime
    return_date: datetime
    status: Optional[str] = "issued"
    class Config:
        json_encoders = {
            ObjectId: str
        }
